src/github.com/matrix-org/dendrite/clientapi/auth/storage/accounts/membership_table.go

```diff
@@ -40,27 +45,34 @@ CREATE UNIQUE INDEX IF NOT EXISTS account_membership_event_id ON account_members
 `
 
 const insertMembershipSQL = `
-	INSERT INTO account_memberships(localpart, room_id, event_id) VALUES ($1, $2, $3)
-	ON CONFLICT (localpart, room_id) DO UPDATE SET event_id = EXCLUDED.event_id
+	INSERT INTO account_memberships(localpart, room_id, event_id, still_in_room)
+	VALUES ($1, $2, $3, $4)
+	ON CONFLICT (localpart, room_id) DO
+	UPDATE SET event_id = EXCLUDED.event_id, still_in_room = EXCLUDED.still_in_room
 `
 
 const selectMembershipSQL = "" +
-	"SELECT * from account_memberships WHERE localpart = $1 AND room_id = $2"
+	"SELECT event_id, still_in_room from account_memberships WHERE localpart = $1 AND room_id = $2"
 
 const selectMembershipsByLocalpartSQL = "" +
-	"SELECT room_id, event_id FROM account_memberships WHERE localpart = $1"
+	"SELECT room_id, event_id FROM account_memberships WHERE localpart = $1 AND still_in_room = true"
+
+const selectMembershipsByRoomIDSQL = "" +
+	"SELECT localpart, event_id FROM account_memberships WHERE room_id = $1 AND still_in_room = true"
```

Comment:
I'm not sure this is the best table to do this sort of query over. It doesn't have an index on the `room_id` so it won't be particularly efficient.

It might be better to use the tables in the roomsever instead?